### PR TITLE
 CLN: Deprecate table_include_index

### LINKED ISSUE #608: Fix table_include_index bug
The class variable `table_include_index`  has a bug in current implementation.
This is the only place this variable is referenced:
https://github.com/equinor/fmu-dataio/blob/6ba4b4903da1c73f3d1a82154df50a727388b540/src/fmu/dataio/dataio.py#L827-L841

It is going in to the `export_file_compute_checksum_md5` as the flag variable if the object is a `pd.Dataframe,` but inside that function the flag is checked against a string instead of a bool.. hence this is always evaluated to False.
https://github.com/equinor/fmu-dataio/blob/6ba4b4903da1c73f3d1a82154df50a727388b540/src/fmu/dataio/_utils.py#L142-L143

Question is, is this functionality actually needed or could we just instruct our users to do a 'reset_index()' on their dataframe if they need this index column preserved?

### PR DESCRIPTION
**Pull request by tnatt**
Closes #608

### CODE REVIEW

**Review by mferrera — APPROVED** [review]
👍

File: src/fmu/dataio/dataio.py [hunk 1]
```diff
@@ -529,6 +527,13 @@ def _show_deprecations_or_notimplemented(self) -> None:
                 "Please remove it from the argument list.",
                 UserWarning,
             )
+        if self.table_include_index:
+            warn(
+                "The 'table_include_index' option is deprecated and has no effect. "
+                "To get the index included in your dataframe, reset the index"
```
Comment (jcrivenaes) [inline src/fmu/dataio/dataio.py:533]: I think you need a blank space after 'index'

**Review by jcrivenaes — APPROVED** [review]
LGTM

File: src/fmu/dataio/dataio.py (hunk 1 above)
Comment (tnatt) [reply src/fmu/dataio/dataio.py:533]: :) yes, well spotted

**Pull request merged**